src/main/java/seedu/mtracker/filemanager/CryptoDecoder.java

```diff
@@ -1,25 +1,57 @@
 package seedu.mtracker.filemanager;
 
+import seedu.mtracker.commons.Validate;
+import seedu.mtracker.error.FileLoadError;
 import seedu.mtracker.model.Instrument;
 import seedu.mtracker.model.InstrumentManager;
 import seedu.mtracker.model.subinstrument.Crypto;
+import seedu.mtracker.ui.TextUi;
 
 import java.time.LocalDate;
 
 public class CryptoDecoder extends InstrumentDecoder {
 
     public static final int CRYPTO_EXPIRY_INDEX = 5;
     public static final int CRYPTO_REMARKS_INDEX = 6;
+    protected static LocalDate decodedExpiry;
+    protected static String decodedRemarks;
+
+    public static boolean isValidExpiry(String[] textSegment) {
+        return Validate.isValidExpiry(textSegment[CRYPTO_EXPIRY_INDEX]);
+    }
+
+    public static void validateAndDecodeSpecificAttributes(String[] textSegment) throws FileLoadError {
+        if (isValidExpiry(textSegment)) {
+            decodeSpecificAttributes(textSegment);
+        }
+        throw new FileLoadError();
+    }
+
+    public static void tryValidateAndDecodeSpecificAttributes(String[] textSegment) {
+        try {
+            validateAndDecodeSpecificAttributes(textSegment);
+        } catch (Exception e) {
+            TextUi.showErrorMessage(e);
+        }
+    }
+
+    private static void decodeSpecificAttributes(String[] textSegment) {
+        decodedExpiry = LocalDate.parse(textSegment[CRYPTO_EXPIRY_INDEX]);
+        decodedRemarks = textSegment[CRYPTO_REMARKS_INDEX];
+    }
```

Comment:
```suggestion
    public static boolean isValidExpiry(String[] textSegment) {
        return Validate.isValidExpiry(textSegment[CRYPTO_EXPIRY_INDEX]);
    }
    
    public String getExpiryFromFile(String[] textSegment) {
         String expiry;
         try {
             expiry = textSegment[CRYTPO_EXPIRY_INDEX];
         } catch (IllegalArgumentException e) {
             throw new InvalidNoSavedExpiryInCryptoError();
         }
         return expiry
    }
    
    public String getRemarksFromFile(String[] textSegment) {
         String remarks;
         try {
             remarks = textSegment[CRYTPO_REMARKS_INDEX];
         } catch (IllegalArgumentException e) {
             throw new InvalidFileRemarksFormatInCryptoError();
         }
         return remarks 
    }

    public static void validateAndDecodeSpecificAttributes(String[] textSegment) throws InvalidNoSavedExpiryInCrypto, InvalidFileRemarksFormatInCryptoError,  InvalidExpirySavedInFileError {
        String expiry = getExpiryFromFile(textSegment);
        String remarks = getRemarksFromFile(textSegment);
        if (!isValidExpiry(expiry)) {
            throw new InvalidExpirySavedInFileError();
        }
        decodeSpecificAttributes(expiry, remarks);
    }

    public static void tryValidateAndDecodeSpecificAttributes(String[] textSegment) {
        try {
            validateAndDecodeSpecificAttributes(textSegment);
        } catch (Exception e) {
            TextUi.showErrorMessage(e);
        }
    }

    private static void decodeSpecificAttributes(String[] textSegment) {
        decodedExpiry = LocalDate.parse(textSegment[CRYPTO_EXPIRY_INDEX]);
        decodedRemarks = textSegment[CRYPTO_REMARKS_INDEX];
    }
```

Reply:
Feel free to abstract out into methods or something if u find out a better implementation